tests_e2e/tests/lib/virtual_machine.py

```diff
@@ -91,11 +91,39 @@ def _begin_restart(self) -> LROPoller:
         Derived classes must provide the implementation for this method using their corresponding begin_restart() implementation
         """
 
+    @abstractmethod
+    def get(self) -> Any:
+        """
+        Retrieves the information about the virtual machine or scale set
+        """
+
+    def create_or_update(self, parameters=None, timeout=5 * 60) -> None:
+        """
+        Creates or updates the virtual machine or scale set with custom settings
+        """
+        if parameters is None:
+            parameters = {}
+
+        log.info("Creating/Updating VM for %s", self._identifier)
+
+        poller: LROPoller = execute_with_retry(lambda: self._begin_create_or_update(parameters))
+
+        poller.wait(timeout=timeout)
+
+        if not poller.done():
+            raise TimeoutError(f"Failed to restart {self._identifier.name} after {timeout} seconds")
```

Comment:
why "Failed to restart"?

Reply:
Good catch, failed to restart doesn't make sense. I'll change